web/src/message_fetch.ts

```diff
@@ -794,3 +807,43 @@ export function initialize(finished_initial_fetch: () => void): void {
         cont: load_more,
     });
 }
+
+export function get_narrow_summary(channel_id: number, topic_name: string): void {
+    const filter = new Filter([
+        {operator: "channel", operand: `${channel_id}`},
+        {operator: "topic", operand: topic_name},
+    ]);
+    const data = {narrow: get_narrow_for_message_fetch(filter)};
+    const channel_name = stream_data.get_stream_name_from_id(channel_id);
+    dialog_widget.launch({
+        html_heading: `Summary of #${channel_name} > ${topic_name}:`,
+        html_body: "",
+        html_submit_button: "Close",
+        close_on_submit: true,
+        on_click() {
+            // Just close the modal, there is nothing else to do.
+        },
+        id: "topic-summary-modal",
+        single_footer_button: true,
+        post_render() {
+            const close_on_success = false;
+            dialog_widget.submit_api_request(
+                channel.get,
+                "/json/messages/summary",
+                data,
+                {
+                    success_continuation(response_data) {
+                        const data = z.object({summary: z.string()}).parse(response_data);
+                        const message = data.summary;
+                        $("#topic-summary-modal .modal__content").html(
+                            render_topic_summary({
+                                message,
+                            }),
+                        );
+                    },
+                },
+                close_on_success,
```

Comment:
This needs basic error-handling so if a server isn't configured it doesn't just hang.

Reply:
![Screenshot from 2025-01-23 21-22-55](https://github.com/user-attachments/assets/d5588748-3ecd-43e7-8433-12c23362f826)

This already does basic error handling. What more are you thinking?

Reply:
Oh, interesting, why I tried it in my development environment, I just got an infinite spinner.